src/main/java/com/fasterxml/jackson/core/io/IOContext.java

```diff
@@ -129,6 +137,30 @@ public IOContext(StreamReadConstraints src, BufferRecycler br,
         _contentReference = contentRef;
         _sourceRef = contentRef.getRawContent();
         _managedResource = managedResource;
+        _maxErrorTokenLength = DEFAULT_MAX_ERROR_TOKEN_LENGTH;
+    }
+
+    /**
+     * Main constructor to use.
+     *
+     * @param src                 constraints for streaming reads
+     * @param br                  BufferRecycler to use, if any ({@code null} if none)
+     * @param contentRef          Input source reference for location reporting
+     * @param managedResource     Whether input source is managed (owned) by Jackson library
+     * @param maxErrorTokenLength Maximum number of characters to include in token reported as part of error messages.                            
+     * @since 2.16
+     */
+    public IOContext(StreamReadConstraints src, BufferRecycler br,
+                    ContentReference contentRef, boolean managedResource, int maxErrorTokenLength)
+    {
+        _streamReadConstraints = (src == null) ?
+                StreamReadConstraints.defaults() : src;
+        _bufferRecycler = br;
+        _contentReference = contentRef;
+        _sourceRef = contentRef.getRawContent();
+        _managedResource = managedResource;
+        _maxErrorTokenLength = maxErrorTokenLength < 0 ? DEFAULT_MAX_ERROR_TOKEN_LENGTH : 
```

Comment:
This behavior should be documented in the Javadoc (i.e. "a negative number to use the the default value")

Reply:
True, true 😄. Applied. Thank you!

Reply:
Looks good to me

Reply:
Cheers